File: details/src/main/java/com/nikita/popularmovies/details/VideosAdapter.java
package com.nikita.popularmovies.details;

import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.facebook.drawee.view.SimpleDraweeView;
import com.nikita.popularmovies.Views;
import com.nikita.popularmovies.common.models.Video;
import com.nikita.popularmovies.common.network.Network;

import java.util.Collections;
import java.util.List;

public final class VideosAdapter extends PagerAdapter {
  private String backdrop = "";
  private List<Video> videos = Collections.emptyList();
  private VideoConsumer videoClickAction;

  public VideosAdapter(VideoConsumer videoClickAction) {
    this.videoClickAction = videoClickAction;
  }

  public interface VideoConsumer {
    void onVideoClick(Video video);
  }

  void changeData(String backdrop, List<Video> videos) {
    this.backdrop = backdrop;
    this.videos = videos;
    notifyDataSetChanged();
  }

  @Override
  public Object instantiateItem(ViewGroup container, int position) {
    View view = LayoutInflater.from(container.getContext()).inflate(R.layout.movie_details_video_item, container, false);

    SimpleDraweeView content = Views.findView(view, R.id.preview);
    final Video video = videos.get(position);
    content.setImageURI(video.site.toLowerCase().equals("youtube") ? ("http://img.youtube.com/vi/" + video.key + "/0.jpg") : Network.posterPathUrl(backdrop));
    container.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        videoClickAction.onVideoClick(video);
      }
    });

    container.addView(view);
    return view;
  }

  @Override
  public void destroyItem(ViewGroup container, int position, Object object) {
    container.removeView((View) object);
  }

  @Override
  public int getCount() {
    return videos.size();
  }

  @Override
  public boolean isViewFromObject(View view, Object object) {
    return view == object;
  }
}

File: details/src/main/kotlin/com/nikita/popularmovies/details/VideosAdapter.kt
package com.nikita.popularmovies.details

import android.support.v4.view.PagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.facebook.drawee.view.SimpleDraweeView
import com.nikita.popularmovies.common.findView
import com.nikita.popularmovies.common.models.Video
import com.nikita.popularmovies.common.network.Network

class VideosAdapter(private var backdrop: String = "",
                    private var videos: List<Video> = emptyList(),
                    private val videoClickAction: (Video) -> Unit) : PagerAdapter() {

  fun changeData(backdrop: String, videos: List<Video>) {
    this.backdrop = backdrop
    this.videos = videos
    notifyDataSetChanged()
  }

  override fun instantiateItem(container: ViewGroup, position: Int): Any {
    val view = LayoutInflater.from(container.context).inflate(R.layout.movie_details_video_item, container, false)

    val preview: SimpleDraweeView = view.findView(R.id.preview)
    val video = videos[position]
    preview.setImageURI(if (video.site.toLowerCase() == "youtube") "http://img.youtube.com/vi/${video.key}/0.jpg" else Network.posterPathUrl(backdrop))
    preview.setOnClickListener({ videoClickAction(video) })

    container.addView(view)
    return view
  }

  override fun destroyItem(container: ViewGroup, position: Int, obj: Any) {
    container.removeView(obj as View)
  }

  override fun isViewFromObject(view: View, obj: Any) = view == obj

  override fun getCount() = videos.size
}